### PR TITLE
WaterCannon done, resource ico

### PR DESCRIPTION
**Pull request by Fire-Head**

### CODE REVIEW

**Comment by erorcun** [comment]
lgtm

File: src/render/WaterCannon.cpp [hunk 1]
```diff
@@ -1,10 +1,320 @@
...
+void CWaterCannon::Init(void)
+{
+	m_nId = 0;
+	m_nCur = 0;
+	m_nTimeCreated = CTimer::GetTimeInMilliseconds();
+	
+	for ( int32 i = 0; i < NUM_SEGMENTPOINTS; i++ )
+		m_abUsed[i] = false;
+	
+	WaterCannonVertices[0].u = 0.0f;
```
Comment (aap) [inline src/render/WaterCannon.cpp]: Use the RW macros for all these.

File: src/render/WaterCannon.cpp [hunk 2]
```diff
@@ -1,10 +1,320 @@
...
+void CWaterCannon::Render(void)
+{
+	RwRenderStateSet(rwRENDERSTATEZWRITEENABLE,      (void *)FALSE);
+	RwRenderStateSet(rwRENDERSTATEVERTEXALPHAENABLE, (void *)TRUE);
+	RwRenderStateSet(rwRENDERSTATEFOGENABLE,         (void *)TRUE);
+	RwRenderStateSet(rwRENDERSTATETEXTURERASTER,     (void *)gpWaterRaster);
+	
+	float v = float(CGeneral::GetRandomNumber() & 255) / 256;
+	
+	WaterCannonVertices[0].v = v;
```
Comment (aap) [inline src/render/WaterCannon.cpp]: Macros.

**Comment by Fire-Head** [comment]
fixed

**Pull request merged**